### ISSUE TITLE
[Feature request]  Relative line number 

### ISSUE THREAD

**Issue by isakhammer** [overleaf/overleaf#979]
Just want to emphasis the issue mentioned [here](https://github.com/overleaf/web/issues/659). I guess every vim user knows the importance of relative line numbers when navigating and would greatly appreciate this feature.

**Comment by adamstuck** [comment]
Agreed this would make the vim bindings much more practical.

**Comment by haakon8855** [comment]
In the meantime while waiting for this feature I made a [chrome extension](https://chrome.google.com/webstore/detail/relative-line-numbering-f/cpndkecejnblehgenlnogffaeekmkphh?hl=no) which runs [these lines of code](https://github.com/overleaf/web/issues/659#issuecomment-571500254) when loading Overleaf to enable relative line numbers. You can check out the code [here](https://github.com/haakon8855/overleafRelativeLineNumbers).

**Comment by crides** [comment]
those js lines apparently need the legacy editor. how do you switch to that? I'm using the "legacy" colorschemes, don't think that necessarily means the editor is legacy though.

**Comment by haakon8855** [comment]
@crides Yeah, it requires the legacy editor (Ace). Unfortunately afaik, overleaf removed the legacy editor entirely __very__ recently. Sadly, I think this means we have to live without relative line numbering in overleaf until either overleaf implements this or the new editor (CodeMirror 6) supports this natively (which is not looking very likely imo).

**Comment by kostyafarber** [comment]
Is there any progress or desire on this one? Looks like [this](https://github.com/jsjoeio/codemirror-line-numbers-relative) could be useful here?

**Comment by InvisOn** [comment]
I can verify this script works. Thank you MattHeffNT!

**Comment by MattHeffNT** [comment]
> Any updates to this issue? I believe that Code Mirror 6 does allow for a relative line numbering option in its setup for Code Mirror 6.x. Here's a thread from the Code Mirror docs from 2021 asking for Vim-style relative line numbering: [Relative (VIM style?) Line Numbers?](https://discuss.codemirror.net/t/relative-vim-style-line-numbers/2855)
> 
> It might be worth exploring whether this functionality is compatible with the current implementation in the Overleaf editor.

Thanks Micah. Once I've got a few projects out of the way I'll have a look and see. Might be a bit cleaner than my userscript. Also if it works I'll open a pull request with Overleaf and see if we can add it as part of their official user settings.
